pkg/tenant/mock/quarterdeck.go

```diff
@@ -0,0 +1,196 @@
+package mock
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+)
+
+const (
+	StatusEP       = "/v1/status"
+	RegisterEP     = "/v1/register"
+	LoginEP        = "/v1/login"
+	AuthenticateEP = "/v1/authenticate"
+	RefreshEP      = "/v1/refresh"
+	APIKeysEP      = "/v1/apikeys"
+)
+
+// Server embeds an httptest Server and provides additional methods for configuring
+// mock responses and counting requests. By default handlers will panic, it's the
+// responsibility of the test writer to configure the behavior of each handler that
+// will be invoked by using the appropriate On* method and passing in the desired
+// HandlerOption(s). If no HandlerOption is specified, the default behavior is to
+// return a 200 OK response with an empty body.
+type Server struct {
```

Comment:
Potentially you might also want to embed an authtest server as well so that you can test authenticated quarterdeck requests?